contracts/modules/cover/Cover.sol

```diff
@@ -402,6 +416,21 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     emit CoverEdited(coverId, cover.productId, lastCoverSegmentIndex + 1, msg.sender);
   }
 
+  function checkFixedPricingAllowedPools(uint productId, PoolAllocationRequest[] memory allocationRequests) internal {
+
+    uint[] memory allowedPools = fixedPricingAllowedPools[productId];
+    for (uint i = 0; i < allocationRequests.length; i++) {
+      bool poolAllowed = false;
+      for (uint j = 0; j < allowedPools.length; i++) {
```

Comment:
note that the second-for loop is incrementing i, instead of j

Reply:
good find